packages/mui-material/src/utils/mergeSlotProps.test.ts

```diff
@@ -0,0 +1,81 @@
+import { expect } from 'chai';
+
+import mergeSlotProps from './mergeSlotProps';
+
+type OwnerState = {
+  className: string;
+  'aria-label'?: string;
+};
+
+describe('utils/index.js', () => {
+  describe('mergeSlotProps', () => {
+    it('external slot props is undefined', () => {
+      expect(
+        mergeSlotProps<OwnerState>(undefined, {
+          className: 'default',
+          'aria-label': 'foo',
+        }),
+      ).to.deep.equal({
+        className: 'default',
+        'aria-label': 'foo',
+      });
+    });
+
+    it('external slot props should override', () => {
+      expect(
+        mergeSlotProps<OwnerState>(
+          { className: 'external' },
```

Comment:
Adding `'aria-label: 'bar'` to the external parameter on all tests would make them more robust as it would catch that in the future, the default parameter always overrides the external parameter.

Reply:
Good point. Updated and added another test for the callback signature.